### DISCUSSION TITLE
goal_position is not a CUDA tensor for own robot, but works for preconfigured robots

### DISCUSSION THREAD

**Discussion by Pavidiran** [NVlabs/curobo#164] [Software Q&A] [answered]
Hi,

I tried to configure our own 6 Dof Robot arm. When running the planner I got the following error:

RuntimeError: goal_position.is_cuda() INTERNAL ASSERT FAILED at "src/curobo/curobolib/cpp/geom_cuda.cpp":213, please report a bug to PyTorch. goal_position must be a CUDA tensor

This error does not appear when I use a different .yml file (ex. franka.yml or ur5e.yml). Thats why I think the bug must be in my yml or urdf file, but I can't find it.

Here is the urdf and yml file:
[urdf_and_yml.zip](https://github.com/NVlabs/curobo/files/14403397/urdf_and_yml.zip)

**Comment by balakumar-s** [comment]
How are you running the planner? Can you share your script?

**Comment by Pavidiran** [comment]
Here is the code. I tried to remove the irrelevent parts. The [full_script.txt](https://github.com/NVlabs/curobo/files/14420215/full_script.txt) is also attached:


          class TrajectoryPlanner(Node):
            """Receive goal pose. Plan trajectory. Call action server of the controller."""
          
            def __init__(self) -> None:
                """Initialise servers and clients. Set up curobo."""
                super().__init__("trajectory_planner")
                self.srv = self.create_service(
                    TargetPose,
                    "/move_eef",
                    self._receive_goal_pose,
                )
          
                self.subscription = self.create_subscription(
                    SensorJointState,
                    "/joint_states",
                    self._current_position,
                    1,
                )
          
                # Curobo setup
                self.tensor_args = TensorDeviceType()
                self.world_file = "collision_table.yml"
                self.robot_file = "tx2_60.yml"
                #self.robot_file = "franka.yml"
          
                self.dof_ = 6
                #self.dof_ = 7
          
                self.motion_gen_config = MotionGenConfig.load_from_robot_config(
                    self.robot_file,
                    self.world_file,
                    self.tensor_args,
                    interpolation_dt=0.01,
                )
          
                self.motion_gen = MotionGen(self.motion_gen_config)
                self.motion_gen.warmup(enable_graph=True)
                self.robot_cfg = load_yaml(
                    join_path(get_robot_configs_path(), self.robot_file),
                )["robot_cfg"]
                self.robot_cfg = RobotConfig.from_dict(self.robot_cfg, self.tensor_args)
                self.retract_cfg = self.motion_gen.get_retract_config()
          
                self.state = self.motion_gen.rollout_fn.compute_kinematics(
                    JointState.from_position(self.retract_cfg.view(1, -1)),
                )
          
                self.q_start = JointState.from_position(
                    self.tensor_args.to_device([[0.0] * self.dof_]),
                )
          
                # World model
                self.world_model = self.motion_gen.world_collision
          
            def _current_position(self, msg: JointState) -> None:
                """Initialize the start position of the robot."""
                start_positions = msg.position[: self.dof_]s also attached
                print(result) # code doesnt reach this point for custom robot
                return self._convert_trajectory(result)
          
            def _create_trajectory_point(
                self, dt: int, i: int, positions: torch.Tensor, joint_names: list,
            ) -> JointTrajectoryPoint:
                """Create trajectory point."""
                point = JointTrajectoryPoint()
                point.time_from_start = Duration(nanosec=dt * i)
                point.positions = [positions[name] for name in range(len(joint_names))]
                return point
          
            def _receive_goal_pose(
                self,
                request: TargetPose.Request,
                response: TargetPose.Response,
            ) -> TargetPose.Response:
                """Receive goal pose. Call actionserver."""
                pose = request.target.pose
                pose_frame_id = request.target.header.frame_id
                # Transform the pose to world frame
               
                goal_position = torch.tensor(
                    [
                        pose.position.x,
                        pose.position.y,
                        pose.position.z,
                    ],
                )
                goal_quaternion = torch.tensor(
                    [
                        pose.orientation.w,
                        pose.orientation.x,
                        pose.orientation.y,
                        pose.orientation.z,
                    ],
                )
                goal_pose = Pose(position=goal_position, quaternion=goal_quaternion)
                goal = self._generate_trajectory(goal_pose)
                response.success = True
                return response
          
          
          def main() -> None:
            """Spin Node."""
            rclpy.init()
            trajectory_planner = TrajectoryPlanner()
            rclpy.spin(trajectory_planner)
            rclpy.shutdown()

**Comment by balakumar-s** [comment, marked as answer]
You need to have position and quaternion on gpu. You can do this by passing `device="cuda"` to your `torch.Tensor()` calls or initialize Pose using `Pose.from_list([x, y, z, qw, qx, qy, qx])`
